rdrf/rdrf/forms/navigation/quick_links.py

```diff
@@ -18,297 +18,257 @@ class QuickLink(object):
+    )
+    RegistrationProfiles = make_link("admin:registration_registrationprofile_changelist", _("Registration Profiles"))
+    LoginLog = make_link("admin:useraudit_loginlog_changelist", _("User Login Log"))
+    FailedLoginLog = make_link("admin:useraudit_failedloginlog_changelist", _("User Failed Login Log"))
+    LoginAttempts = make_link("admin:useraudit_loginattempt_changelist", _("User Login Attempts Log"))
+    IPRestrictRangeGroup = make_link("admin:iprestrict_rangebasedipgroup_changelist", _("IP Restrict Groups"))
+    IPRestrictGeoGroup = make_link("admin:iprestrict_locationbasedipgroup_changelist", _("IP Restrict Geolocations"))
+    IPRestrictRule = make_link("admin:iprestrict_rule_changelist", _("IP Restrict Rules"))
+    Sites = make_link("admin:sites_site_changelist", _("Sites"))
+    ParentGuardian = make_link("admin:patients_parentguardian_changelist", _("Parents/Guardians"))
 
-    def __new__(cls):
-        if not MenuConfig._instance:
-            MenuConfig._instance = MenuConfig.MenuConfigInner()
-        return MenuConfig._instance
+    DemographicsFields = make_link("admin:rdrf_demographicfields_changelist", _("Registry Demographics Fields"))
+    ConsentRules = make_link("admin:rdrf_consentrule_changelist", _("Consent Rules"))
+    Surveys = make_link("admin:rdrf_survey_changelist", _("Surveys"))
+    SurveyAssignments = make_link("admin:rdrf_surveyassignment_changelist", _("Survey Assignments"))
+    SurveyRequest = make_link("admin:rdrf_surveyrequest_changelist", _("Survey Request"))
 
-    def __getattr__(self, name):
-        return getattr(self._instance, name)
+    RegistryForms = make_link("admin:rdrf_registryform_changelist", _("Registry Forms"))
+    Sections = make_link("admin:rdrf_section_changelist", _("Registry Sections"))
+    DataElements = make_link("admin:rdrf_commondataelement_changelist", _("Registry Common Data Elements"))
+    PermissibleValueGroups = make_link(
+        "admin:rdrf_cdepermittedvaluegroup_changelist",
+        _("Registry Permissible Value Groups")
+    )
+    PermissibleValues = make_link("admin:rdrf_cdepermittedvalue_changelist", _("Registry Permissible Values"))
+    ConsentSections = make_link("admin:rdrf_consentsection_changelist", _("Registry Consent Sections"))
+    ConsentValues = make_link("admin:patients_consentvalue_changelist", _("Registry Consent Values"))
+    CdePolicy = make_link("admin:rdrf_cdepolicy_changelist", _("Registry Common Data Elements Policy"))
+    ContextFormGroups = make_link("admin:rdrf_contextformgroup_changelist", _("Registry Context Form Groups"))
 
-    def __setattr__(self, name):
-        return setattr(self._instance, name)
 
+class Links:
+    """
+    All links that can appear in menus are defined.
+    Links are also grouped into related functional areas to make for easier assignment to menus
+    """
+    # related links are grouped or convenience
+    AUDITING = dict([
```

Comment:
Could be a tad nicer if the `dict([make_entry(QL1), ..., make_entry(QLN)])` construct would be:
`links(QL1, ... , QLN)`. etc.

Reply:
Updated.